checks.d/sidekiq.py

```diff
@@ -0,0 +1,261 @@
+'''
+Sidekiq checks
+'''
+# stdlib
+import time
+
+# 3rd party
+import redis
+
+# project
+from checks import AgentCheck
+
+class Stats:
+    def __init__(self, conn, namespace):
+        self.conn = conn
+        self.namespace = namespace
+        self._script = self._register_script()
+        self._check_stats()
+
+    # This script collects all the sidekiq related stats in one go.
+    # KEYS[1] = 'stat:processed'
+    # KEYS[2] = 'stat:failed'
+    # KEYS[3] = 'schedule'
+    # KEYS[4] = 'retry'
+    # KEYS[5] = 'dead'
+    # KEYS[6] = 'processes'
+    # KEYS[7] = 'queues'
+    # KEYS[8] = 'queue:*'
+    #
+    # Returns an array of results. The array follows the following format:
+    # [
+    #   processed,
+    #   failed,
+    #   scheduled_size,
+    #   retry_size,
+    #   dead_size,
+    #   processes_size,
+    #   workers_size,
+    #   number_of_queues,
+    #   queue name 1,
+    #   queue name 2,
+    #   ...,
+    #   queue name n,
+    #   queue length 1,
+    #   queue length 2,
+    #   ...,
+    #   queue length n,
+    #   queue latency 1,
+    #   queue latency 2,
+    #   ...,
+    #   queue latency n
+    # ]
+    #
+    def _register_script(self):
+        s = """
+        local processed = redis.call("GET", KEYS[1])
```

Comment:
These `GET` calls could possibly be switched to `MGET` calls in order to save roundtrips.